src/DDTrace/Tracer.php

```diff
@@ -430,42 +409,26 @@ private function setPrioritySamplingFromSpan(Span $span)
         }
 
         $prioritySampling = $span->getContext()->getPropagatedPrioritySampling();
-        if (null === $prioritySampling) {
-            $prioritySampling = $this->sampler->getPrioritySampling($span);
+        if (null !== $prioritySampling) {
+            $this->setPrioritySampling($prioritySampling);
         }
-        $this->setPrioritySampling($prioritySampling);
     }
 
     /**
      * @param mixed $prioritySampling
      */
     public function setPrioritySampling($prioritySampling)
     {
-        $this->prioritySampling = $prioritySampling;
-
-        $rootSpan = $this->getSafeRootSpan();
-        if (null === $rootSpan) {
-            return;
-        }
-
-        if ($prioritySampling !== Sampling\PrioritySampling::UNKNOWN) {
-            $rootSpan->setMetric('_sampling_priority_v1', $prioritySampling);
-        } elseif (isset($rootSpan->metrics)) { // official API does not allow removal, so...
-            unset($rootSpan->metrics['_sampling_priority_v1']);
-        }
+        set_priority_sampling($prioritySampling);
+        set_priority_sampling($prioritySampling, true);
```

Comment:
(This might be related to what you mentioned in the PR description, "priority_sampling is not yet detached from the root_span".) It looks like we have two sources of truth for the sampling priority value.

The request global `default_priority_sampling` is initially set from a distributed trace context (before there is a root span). We are assuming the root span to be the source of truth for priority sampling but if there isn't a root span, the request global is the priority sampling source of truth. 

The request global gets updated on this second call to `set_priority_sampling`. (I'm assuming this also means that if the tracer flushes the trace and starts a new trace during the same request, the sampling priority will be that of the previous trace, not the original value from the distributed tracing context.)

Are we planning on eventually migrating away from using the root span as the source of truth for priority sampling and rely exclusively on the request global for that value? If that is the case, can we reduce this to one call to `set_priority_sampling` and just set the request global with the call?

Reply:
I do not know yet. This is part of manual tracing discussion.
But I think that we'll need a default and an explicit decision (like "drop the sampling of this connection" like in an async connection handler where you don't want to sample everything).